internal/api/resources_test.go

```diff
@@ -404,13 +405,40 @@ func TestPutResource(baseT *testing.T) {
 	})
 }
 
+func TestMaxAssetSizeFor(t *testing.T) {
+	var (
+		maxBarSize = p2uint64(42)
+		maxFooSize = p2uint64(30)
+	)
+
+	cfg := core.Config{
+		MaxAssetSize: []core.MaxAssetSizeRule{
+			{
+				Regex: regexp.MustCompile("foo.*"),
+				Value: maxFooSize,
+			},
+			{
+				Regex: regexp.MustCompile(".*bar"),
+				Value: maxBarSize,
+			},
+		},
+	}
+
+	assert.DeepEqual(t, "foo", cfg.MaxAssetSizeFor(db.AssetType("foo")), maxFooSize)
+	assert.DeepEqual(t, "bar", cfg.MaxAssetSizeFor(db.AssetType("bar")), maxBarSize)
+	assert.DeepEqual(t, "buz", cfg.MaxAssetSizeFor(db.AssetType("buz")), (*uint64)(nil))
```

Comment:
Please add a test where multiple regexes match at the same time.

Reply:
Oh, another thing. Please also add a test to check the `^(?:...)$` behavior, e.g. `db.AssetType("somefoo")` should not have any limit applied even though it matches `foo` because it does not match `^(?:foo)$`.